base-notebook/Dockerfile

```diff
@@ -47,9 +47,12 @@ ENV PATH=$CONDA_DIR/bin:$PATH \
 ADD fix-permissions /usr/local/bin/fix-permissions
 # Create jovyan user with UID=1000 and in the 'users' group
 # and make sure these dirs are writable by the `users` group.
+# Make /etc/passwd and /etc/group writable so can add entries
+# for uid/gid when container run as arbitrarily assigned uid.
 RUN useradd -m -s /bin/bash -N -u $NB_UID $NB_USER && \
     mkdir -p $CONDA_DIR && \
     chown $NB_USER:$NB_GID $CONDA_DIR && \
+    chmod g+w /etc/passwd /etc/group && \
```

Comment:
This makes it trivial for any user to become root in their container:

```bash
cp /etc/passwd /tmp/passwd
rootpass=`openssl passwd -1 root`
cat /tmp/passwd | sed "s/root:x/root:${rootpass}/" > /etc/passwd
```

Now the user can `su root` with the password 'root'

I think we need another solution that doesn't grant universal `su root` access.

This does only affect putting users into the root group, so not the default case, but it does mean that putting users into the root group is equivalent to letting the users run as root. Is that the intended effect?

Reply:
I will check, but I don't believe that will work in secure environments like Kubernetes/OpenShift where you are blocked from running setuid executables and various related system calls behind the ability to change real user/group.

Reply:
Yep, this is restricted in Kubernetes/OpenShift.

```
jovyan@image:~$ su root
Password:
initgroups: Operation not permitted
```

This is because secure container environments will drop capabilities.

```
                "securityContext": {
                    "capabilities": {
                        "drop": [
                            "KILL",
                            "MKNOD",
                            "SETGID",
                            "SETUID"
                        ]
                    },
                    "privileged": false,
                    "runAsUser": 1000030000,
                    "seLinuxOptions": {
                        "level": "s0:c6,c0"
                    }
                },
```